odoo/api.py

```diff
@@ -632,24 +632,44 @@ def __call__(self, cr=None, user=None, context=None, su=None):
         return Environment(cr, uid, context, su, self.uid_origin)
 
     def ref(self, xml_id, raise_if_not_found=True):
```

Comment:
You could define just one function:
`def ref(self, *xml_id: str, raise_if_not_found: bool) -> BaseModel | None`

I you don't have *exactly one* model, it would raise an exception.

```python
model_name = None
ids = []
for xml_id in xml_ids;
  res_model, res_id = ...
  if not (res_model and res_id):
    continue
  if model_name != res_model:
    if model_name is not None:
      raise ValueError(...)
    model_name = res_modeld
  ids.append(res_id)
return self[model_name].browse(ids) if model_name else None
```

Reply:
That's much better, I like that idea.  It is 100% backward-compatible and fully consistent.

Reply:
I like this too 👍🏻 

Most grouped `ref` calls are actually for the same model anyway, and you can still loop and do one call per model. The query count would not change.